lib/bolt/cli.rb

```diff
@@ -858,77 +873,57 @@ def initialize_project
-        config.project.path + '.modules',
-        config.project.modules
-      )
+      moduledir = project.path + '.modules'
+      installer = Bolt::ModuleInstaller.new(outputter, pal)
+
+      installer.install(project.modules, project.puppetfile, moduledir, force: force)
     end
 
-    # Installs modules declared in the project configuration file.
+    # Adds a single module to the project.
     #
-    def install_modules(puppetfile_path, config, moduledir, modules)
-      require 'bolt/puppetfile'
-      require 'bolt/puppetfile/installer'
-
-      puppetfile = Bolt::Puppetfile.new(modules)
-
-      # If the Puppetfile exists, check if it includes specs for each declared
-      # module, erroring if there are any missing. Otherwise, resolve the
-      # module dependencies and write a new Puppetfile. Users can forcibly
-      # overwrite an existing Puppetfile with the '--force' option.
-      if puppetfile_path.exist? && !options[:force]
-        outputter.print_message "Parsing existing Puppetfile at #{puppetfile_path}"
-        existing = Bolt::Puppetfile.parse(puppetfile_path)
-
-        unless existing.modules.superset? puppetfile.modules
-          missing_modules = puppetfile.modules - existing.modules
-
-          raise Bolt::Error.new(
-            "Puppetfile #{puppetfile_path} is missing specifications for modules: "\
-            "#{missing_modules.map(&:title).join(', ')}. This may not be a Puppetfile "\
-            "managed by Bolt. To forcibly overwrite the Puppetfile, run with the "\
-            "'--force' option.",
-            'bolt/missing-module-specs'
-          )
-        end
-      else
-        outputter.print_message "Resolving module dependencies, this may take a moment"
-        puppetfile.resolve
-        outputter.print_message "Writing Puppetfile at #{puppetfile_path}"
-        puppetfile.write(puppetfile_path, force: true)
-      end
+    def add_project_module(name, project)
+      require 'bolt/module_installer'
 
-      outputter.print_message "Syncing modules from #{puppetfile_path} to #{moduledir}"
-      ok = Bolt::Puppetfile::Installer.new(config).install(puppetfile_path, moduledir)
+      assert_project_file(project)
 
-      # Automatically generate types after installing modules.
-      pal.generate_types
+      moduledir = project.path + '.modules'
+      modules   = project.modules || []
+      installer = Bolt::ModuleInstaller.new(outputter, pal)
 
-      outputter.print_puppetfile_result(ok, puppetfile_path, moduledir)
-      ok ? 0 : 1
+      installer.add(name, modules, project.puppetfile, moduledir, project.project_file)
+    end
+
+    # Asserts that there is a project configuration file.
+    #
+    def assert_project_file(project)
+      unless project.project_file?
+        raise Bolt::Error.new(
+          "Could not find project configuration file #{project.project_file}, unable "\
+          "to install modules. To create a Bolt project with a project configuration "\
```

Comment:
```suggestion
          "to install modules. To create a Bolt project run 'bolt project init'."\
```

Reply:
This just seems a little verbose...`bolt project init` won't actually make a bolt-project.yaml if bolt.yaml exists, so I don't think we need to worry about differentiating that. If we really want to make it a good experience, we could have a separate message about renaming bolt.yaml to bolt-project.yaml if it exists.